tests/bsim/bluetooth/host/adv/chain/pytest/test_adv_chain.py

```diff
@@ -0,0 +1,24 @@
+# Copyright (c) 2023 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+
+from pathlib import Path
+from twister_harness.helpers.babblesim import BabbleSim
+
+
+logger = logging.getLogger(__name__)
+
+
+def test_adv_chain(bsim_env: dict, build_dir: Path):
+    new_app_name = 'bs_nrf52_bsim_tests_bsim_bluetooth_host_adv_chain_prj_conf'
+    script_name = 'adv_chain.sh'
+    script_path = Path(__file__).resolve().parent / script_name
+    timeout = 10.0  # [s]
+
+    bsim = BabbleSim()
+    bsim.copy_app(bsim_env, build_dir, new_app_name)
```

Comment:
Yes, it would be great to avoid coping binaries in test itself. As I wrote in first comment, I believe that we could be able to avoid doing this if we rewrite test execution from bash scripts to Python. Alternatively, we could consider adding somehow this coping step to building process.

To avoid overwriting the same binaries, I planned to group all tests, which use the same binary under one pytest module (or class – please look at `host/adv/periodic/pytest/test_adv_periodic.py` file) and thanks to this ensure that coping will be performed only once. This is not a perfect solution due to potential problem with parallelization of execution of those pytest tests among one module or class. Eventually, I can write more complicated mechanism which will keep information in file (protected by filelock) about performed coping actions – if test discovers, that coping was already performed, then coping will be skipped